service/controller/v14/resource/endpoint/create.go

```diff
@@ -32,36 +37,40 @@ func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desir
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
-	if currentEndpoint != nil {
-		return nil, nil // An endpoint exists already, we should update instead of create.
-	}
-
 	desiredEndpoint, err := toEndpoint(desiredState)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
-	if desiredEndpoint == nil {
-		return nil, nil // Nothing to do.
-	}
```

Comment:
Removed the desired state checks. We define the desired state. It must be there. If it is not, we fucked up very very badly. Below I also removed the test cases for this.